var/spack/repos/builtin/packages/py-gevent/package.py

```diff
@@ -38,3 +38,10 @@ class PyGevent(PythonPackage):
 
     # Deprecated compiler options. upstream PR: https://github.com/gevent/gevent/pull/1896
     patch("icc.patch", when="%intel")
+
+    def flag_handler(self, name, flags):
+        iflags = []
+        if name == "cflags":
+            if self.spec.satisfies("%oneapi@2023.0.0:"):
+                iflags.append("-Wno-error=incompatible-function-pointer-types")
+        return (iflags, None, None)
```

Comment:
```suggestion
        if name == "cflags":
            if self.spec.satisfies("%oneapi@2023:"):
                flags.append("-Wno-error=incompatible-function-pointer-types")
        return (flags, None, None)
```
I believe `flags` may be pre-populated by things like `compilers.yaml` so you want to append instead of override.